packages/select/src/components/query-list/queryList.tsx

```diff
@@ -229,10 +275,15 @@ export class QueryList<T> extends AbstractComponent<QueryListProps<T>, QueryList
                 ...spreadableState,
                 items,
                 itemsParentRef: this.refHandlers.itemsParent,
-                menuProps,
+                menuProps: {
+                    ...menuProps,
+                    id: this.listId,
```

Comment:
Won't setting an `id` here replace the id referenced by the outermost combobox that already defines `aria-controls`? I'm a bit confused as to what we're trying to accomplish. Here's a before/after of these changes on a simple example to illustrate:

| Before | After |
|--------|--------|
| <img width="2620" height="1436" alt="before" src="https://github.com/user-attachments/assets/4554c17c-5903-47a4-90f0-65aafbf61a6f" /> | <img width="2436" height="1436" alt="after" src="https://github.com/user-attachments/assets/d497b505-3bd7-43a0-be3b-c7f87c88fe36" /> |

https://codesandbox.io/p/sandbox/v4x4r8